lib/phoenix_live_view/socket.ex

```diff
@@ -82,4 +93,53 @@ defmodule Phoenix.LiveView.Socket do
 
   @impl Phoenix.Socket
   def id(socket), do: socket.private.connect_info[:session]["live_socket_id"]
+
+  defmacro __using__(_opts) do
+    quote do
+      use Phoenix.Socket
+
+      channel "lvu:*", Phoenix.LiveView.UploadChannel
+      channel "lv:*", Phoenix.LiveView.Channel
+
+      @phoenix_connect false
+      @phoenix_id false
+
+      @before_compile unquote(__MODULE__)
+      @on_definition unquote(__MODULE__)
+    end
+  end
+
+  def __on_definition__(env, :def, :connect, [_, _, _], _guards, _body) do
+    Module.put_attribute(env.module, :phoenix_connect, true)
```

Comment:
kind of a kludge @josevalim , but I think this is the only decent way to remove user responsibility of calling our own `connect/3`?